test/e2e/ginkgo_operator-sdk.go

```diff
@@ -0,0 +1,230 @@
+	os.Args = append(os.Args, "-root", rootDir, "-singleNamespace", "-globalMan", globalManFile, "-namespacedMan", namespacedManFile)
+
+	return nil
+}
+
+func cleanupOperatorSDK() error {
+	// determine the ginkgo seed
+	ginkgoSeed, err := getGinkoSeed()
+	if err != nil {
+		return err
+	}
+
+	globalManFile := fmt.Sprintf("%s/global-manifest-%s.yaml", os.TempDir(), ginkgoSeed)
+	err = os.Remove(globalManFile)
+	if err != nil {
+		return err
+	}
+
+	namespacedManFile := fmt.Sprintf("%s/namespace-manifest-%s.yaml", os.TempDir(), ginkgoSeed)
+	err = os.Remove(namespacedManFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getArg(argName string) (bool, string) {
+	for i, arg := range os.Args {
+		if arg == argName {
+			return true, os.Args[i+1]
+		} else if strings.HasPrefix(arg, argName+"=") {
+			argAndValue := strings.SplitN(arg, "=", 2)
+			return true, argAndValue[1]
+		}
+	}
+	return false, ""
+}
+
+func getGinkgoNode() int {
+	defined, ginkgoNodeString := getArg("--ginkgo.parallel.node")
+	if !defined {
+		return 1
+	}
+	ginkgoNode, err := strconv.Atoi(ginkgoNodeString)
+	if err != nil {
+		fmt.Printf("Error: %s", err.Error())
+		return 0
+	}
+	return ginkgoNode
+}
+
+func getGinkoSeed() (string, error) {
+	defined, ginkgoSeed := getArg("--ginkgo.seed")
+	if !defined {
+		return "", errors.New("unable to find --ginko.seed argument")
+	}
+	return ginkgoSeed, nil
+}
+
+func getRootDir() (string, error) {
+	rootDir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	for info, err := os.Stat(rootDir + "/LICENSE"); os.IsNotExist(err) || (err == nil && info.IsDir()); {
+		rootDir = filepath.Dir(rootDir)
+		if rootDir == "/" {
+			return "", errors.New("Failed to determine root directory")
+		}
+		info, err = os.Stat(rootDir + "/LICENSE")
+	}
+
+	return rootDir, nil
+}
+
+func isGinkgoRunner() bool {
+	defined, _ := getArg("--ginkgo.seed")
+	return defined
+}
+
+func populateFrameworkScheme() error {
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, &operator.BuildList{})
+	if err != nil {
+		Logf("Failed to add BuildList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, &operator.BuildStrategyList{})
+	if err != nil {
+		Logf("Failed to add BuildStrategyList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, &operator.ClusterBuildStrategyList{})
+	if err != nil {
+		Logf("Failed to add ClusterBuildStrategyList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(pipelinev1.AddToScheme, &pipelinev1.TaskList{})
+	if err != nil {
+		Logf("Failed to add TaskList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(pipelinev1.AddToScheme, &pipelinev1.TaskRunList{})
+	if err != nil {
+		Logf("Failed to add TaskRunList to schema %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func startLocalOperator() (*exec.Cmd, error) {
```

Comment:
trying to understand the reasoning behind calling the cli, and not the previous defined `WaitForOperatorDeployment` func, when starting the operator.

Reply:
The implementation of `WaitForOperatorDeployment` uses the k8s API to check if deployment exists that represents the operator-sdk, see https://github.com/operator-framework/operator-sdk/blob/master/pkg/test/e2eutil/wait_util.go#L55. So, `WaitForOperatorDeployment` does not start something.

This wait call is still in `SynchronizedBeforeSuite`, but its logic currently does not run because we have a locally running operator and `framework.Global.LocalOperator` is set to true. We will need to add another environment variable soon to skip the local operator startup for those scenarios where the e2e tests should run against a ready-configured k8s cluster where the operator is deployed already.

But to retain the capability to start the locally running operator, especially for developing purposes = to run the e2e tests locally after having made code changes to the operator runtime codebase, I think we need this.

Reply:
Discussed with @qu1queee; we leave the code as it is. One advantage of running the operator using a command is that one can collect all the stdout/err output and print it out at the end.